utils-intervalrdd/src/main/scala/org/bdgenomics/utils/rangearray/Interval.scala

```diff
@@ -16,14 +16,14 @@
  * limitations under the License.
  */
 
-package org.bdgenomics.utils.intervaltree
+package org.bdgenomics.utils.rangearray
 
 /**
  * An interval is a 2-dimensional 0-based coordinate consisting of a closed start value and open end value.
  * Each 2-dimensional coordinate has a defined width.
  * This can be used to express a region of a genome, a transcript, a gene, etc.
  */
-trait Interval {
+trait Interval[T <: Interval[T]] extends Comparable[T] {
```

Comment:
I don't think this type needs to be recursive. I think the `overlaps` and `compareTo` functions should take params of type `Interval` and not `T`. However, is this necessary to support `Interval`s that exist on different coordinate planes? I.e., going from supporting a single contig to multiple contigs? If so, I can get behind the recursive type.